### ISSUE TITLE
Not all the URL must have "//"

### ISSUE THREAD

**Issue by ibc** [unshiftio/url-parse#8]
This is also a URL: `sip:alice@atlanta.com`. But:

```
javascript
var URL = require('url-parse')

new URL('sip:alice@atlanta.com')
{ hash: '',
  query: '',
  protocol: '',
  pathname: '',
  auth: 'sip:alice',
  host: 'atlanta.com',
  port: '',
  hostname: 'atlanta.com',
  password: 'alice',
  username: 'sip',
  href: '//sip:alice@atlanta.com' }
```

Just no. It should be:

```javascript
{ hash: '',
  query: '',
  protocol: 'sip',
  pathname: '',
  auth: '',
  host: 'atlanta.com',
  port: '',
  hostname: 'atlanta.com',
  password: '',
  username: 'alice',
  href: '' }
```

If this library is just supposed to parse [Common Internet Scheme Syntax](http://tools.ietf.org/html/rfc1738#section-3.1) please specify it in the doc.

**Comment by spyworldxp** [comment]
Just path only.

```
<a href="/fdsfdsfd/dfd">

URL {
  hash: '',
  query: {},
  protocol: '',
  pathname: '/fdsfdsfd/dfd',
  auth: '',
  host: '',
  port: '',
  hostname: '',
  password: '',
  username: '',
  href: '///fdsfdsfd/dfd' }
```

**Comment by hongkongkiwi** [comment]
Same problem here.

**Comment by ibc** [comment]
A path is not a URL.

**Comment by lpinca** [comment]
Running the lib from this page with the relative URL specified above:
```js
URLParse('/fdsfdsfd/dfd');

URL {hash: "", query: "", protocol: "https:", pathname: "/fdsfdsfd/dfd", auth: ""…}

auth: ""hash: ""
host: "github.com"
hostname: "github.com"
href: "https://github.com/fdsfdsfd/dfd"
password: ""
pathname: "/fdsfdsfd/dfd"
port: ""
protocol: "https:"
query: ""
username: ""
```
The library needs to know the `baseURL` when the URL to parse is a relative URL.
`baseURL` defaults to [`location`](https://developer.mozilla.org/en-US/docs/Web/API/Location) in the browser.

**Comment by ibc** [comment]
HTTP URL != URL

I expected this to be a generic URL parser rahter than a URL parser just for web stuff.

**Comment by lpinca** [comment]
@ibc that's a valid point and your original report is open for that reason, but the relative URL is a different issue IMO.

The browser throws an error if you don't specify a `baseURL`:
```js
var url = new URL('/fdsfdsfd/dfd'); // Throws 
```
We try use a sane default value.

**Comment by ibc** [comment]
@lpinca makes sense. Thanks for the explanation.

**Comment by lpinca** [comment]
Fixed by #22, thank you @iamtheib.

**Comment by ibc** [comment]
:cake:
